tests/integ/test_dataframe.py

```diff
@@ -1462,6 +1462,24 @@ def test_df_col(session):
     assert isinstance(c._expression, Star)
 
 
+@pytest.mark.xfail(
+    "config.getoption('local_testing_mode', default=False)",
+    reason="Session.sql and DataFrame.cache_result not supported",
```

Comment:
@sfc-gh-aling is that the state of today for cache_result?

Reply:
there are two limitations in local testing which is why we cannot enable this test for local testing.

1. `session.sql` is not supported. We use the sql query to test that this sql is present in query history.
2. even if we replace `session.sql` with `session.create_dataframe` to create the test dataframe, query listeners are not supported in local testing since we do not build sql queries in local testing mode.

since this test is specifically testing the queries generated by cache_result, it should be fine to skip this test for local testing.

Reply:
From the above description, it seems the reason describe is not accurate, where the reason seems because session.sql, and session.query_history is not supported with local testing today, not because of cache_result.
How about let's create a dataframe use session.create_dataframe here, then if the local testing is able to support query_history, they can turn this test on. Also, update the reason to reflect the actual reason.